sdk/core/core/src/az_span_builder.c

```diff
@@ -23,6 +25,59 @@ AZ_NODISCARD az_result az_span_builder_append_byte(az_span_builder * const self,
   return AZ_OK;
 }
 
+AZ_INLINE uint8_t _az_decimal_to_ascii(uint8_t const d) {
+  assert(d < 10);
+  return '0' + d;
+}
+
+AZ_INLINE AZ_NODISCARD az_result
+_az_span_builder_append_unsigned_number(az_span_builder * const self, unsigned long long const n) {
+  if (n == 0) {
+    return az_span_builder_append_byte(self, '0');
+  }
+
+  static unsigned long long div_max = 1;
```

Comment:
```suggestion
  static uint64_t div_max = 1;
```

Reply:
Should we end up using `int*_t`, we then know `div_max` right away, and no need to calculate it. But think we should use `unsigned long long`, and not `uint64_t`.

Reply:
Most of the time, when we write/read data, we would like to be as much deterministic as possible. There are only few platform-dependant things that we use and, usually, they temporary and not serializable. For example, pointers, `size_t` (a size of array in memory), `ptrdiff_t` (a difference between pointers). Everything else most-likely should have constant size on all platforms.

Reply:
Done.